src/Microsoft.DncEng.SecretManager/SecretTypes/GitHubBotAccount.cs

```diff
@@ -0,0 +1,205 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.DncEng.CommandLineLib;
+
+namespace Microsoft.DncEng.SecretManager.SecretTypes
+{
+    [Name("github-account")]
+    public class GitHubBotAccount : SecretType<GitHubBotAccount.Parameters>
+    {
+        const string PasswordSuffix = "-password";
+        const string SecretSuffix = "-secret";
+        const string RecoveryCodesSuffix = "-recovery-codes";
+
+        private readonly List<string> _suffixes = new List<string> { PasswordSuffix, RecoveryCodesSuffix, SecretSuffix };
+        private readonly ISystemClock _clock;
+        private readonly IConsole _console;
+
+        public class Parameters
+        {
+            public string Name { get; set; }
+        }
+
+        public GitHubBotAccount(ISystemClock clock, IConsole console)
+        {
+            _clock = clock;
+            _console = console;
+        }
+
+        public override List<string> GetCompositeSecretSuffixes()
+        {
+            return _suffixes;
+        }
+
+        public override async Task<List<SecretData>> RotateValues(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            if (!_console.IsInteractive)
+            {
+                throw new InvalidOperationException($"User intervention required for creation or rotation of GitHub bot account.");
+            }
+
+            string password = await context.GetSecretValue(context.SecretName + PasswordSuffix);
+            if (string.IsNullOrEmpty(password))
+            {
+                return await NewAccount(parameters, context, cancellationToken);
+            }
+            else
+            {
+                return await UpdateAccount(password, parameters, context, cancellationToken);
+            }
+        }
+
+        private async Task<List<SecretData>> UpdateAccount(string password, Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            _console.WriteLine($"Please login to GitHub account {parameters.Name} using password {password}");
```

Comment:
I initially wanted to copy password into clipboard, but problem is that if we want to do it platform independent then it will take more work.

Reply:
sounds good.